code/obj/item/cigarette.dm

```diff
@@ -1121,32 +1124,34 @@
 		user.visible_message("<span style='color:red'><b>[user]</b> waves [src] around in front of [target]'s face! OoOo, are ya scared?![src.on ? "" : " No, probably not, since [src] is closed."]</span>")
 		return
 
-	afterattack(atom/target, mob/user as mob)
-		if (!on && istype(target, /obj/reagent_dispensers/fueltank))
-			if (src.fuel == -1)
+	afterattack(atom/O, mob/user as mob)
+		if (!on && (istype(O, /obj/reagent_dispensers/fueltank) || istype(O, /obj/item/reagent_containers/food/drinks/fueltank)))
+			if (infinite_fuel)
 				user.show_text("You can't seem to find any way to add more fuel to [src]. It's probably fine.", "blue")
 				return
-			var/obj/reagent_dispensers/fueltank/O = target
-			var/fuelamt = O.reagents.get_reagent_amount("fuel")
-			if (fuelamt)
-				var/removed = min(fuelamt, 50)
-				O.reagents.remove_reagent("fuel", removed)
-				fuel += removed
-				user.show_text("[src] refueled.", "blue")
-				playsound(user.loc, "sound/effects/zzzt.ogg", 50, 1, -6)
+
+			if (get_fuel() >= src.reagents.maximum_volume) //this could be == but just in case...
+				boutput(user, "<span style='color:red'>[src] is full!</span>")
+				return
+
+			if (O.reagents.total_volume)
+				O.reagents.trans_to(src, src.reagents.maximum_volume)
```

Comment:
No worries, I appreciate the feedback.

I got a question though. I started wondering, if reagents are automatically added in the beginning and we're already checking the amount of reagent, is there even a point in checking for `if (reagents)`? I copied from the welder code so I kinda just put it in without thinking, but looking at it again it seems kinda pointless to me.

Reply:
It's a little pointless because `reagents` will basically always exist; like you said, it's automatically created on `New()`. However, it's good practice to have checks like that, just in case (though that's more relevant to other situations).